### PR TITLE
Remove `bytes_be_to_int` and `int_to_bytes_be`, static assert size of `int` and `unsigned`, refactoring

### PR DESCRIPTION
**Pull request by Robyt3**
Supersedes #6263.

## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
Why did you change `4` to `sizeof(int)`? `sizeof(int)` seems more indirect than `4` to me. The places you changed seem like file formats; even if `int` changed, it'd still be 4 bytes on disk/over the network/etc.

**Comment by Robyt3** [comment]
I prefer using `sizeof(int)` to signify that the `char` arrays are used for packing `int`/`unsigned`. If `int` or `unsigned` ever change than a lot of stuff breaks everywhere in any case. That's why I added the static assert to ensure that the size is 4.

**Comment by heinrich5991** [comment]
But to me, it seems it's not `int`s/`unsigned`s getting packed, but 32-bit (4-byte) integers.

**Comment by Robyt3** [comment]
I could also use `sizeof(int32_t)` instead if that makes it clearer that those are 4 byte arrays representing 32-bit integers.

File: src/base/system.cpp [hunk 1]
```diff
@@ -3838,33 +3838,8 @@ const char *str_next_token(const char *str, const char *delim, char *buffer, int
...
-	{
-#if defined(CONF_ARCH_ENDIAN_BIG)
-		bytes[i] = pValue[i];
-#else
-		bytes[sizeof(int) - i - 1] = pValue[i];
-#endif
-	}
-}
+static_assert(sizeof(unsigned) == 4, "unsigned must be 4 bytes in size");
+static_assert(sizeof(unsigned) == sizeof(int), "unsigned and int must have the same size");
```
Comment (heinrich5991) [inline src/base/system.cpp:3842]: I think this is guaranteed by the C standard, but even if it is, it doesn't hurt to have this assertion.

**Review by heinrich5991 — APPROVED** [review]
bors r+

Thanks!

**Pull request merged**